eam-light-backendweb/src/main/java/ch/cern/cmms/eamlightweb/workorders/partusage/PartUsageRest.java

```diff
@@ -96,6 +101,64 @@ public Response loadBinList(@QueryParam("transaction") String transaction, @Quer
 		}
 	}
 
+	@GET
+	@Path("/lots")
+	@Produces("application/json")
+	@Consumes("application/json")
+	public Response loadLotList(@QueryParam("transaction") String transaction, @QueryParam("lot") String lot,
+								@QueryParam("bin") String bin, @QueryParam("part") String part,
+								@QueryParam("store") String store,
+								@QueryParam("requireAvailableQty") boolean requireAvailableQty) {
+		try {
+			GridRequest gridRequest;
+			InforContext context = authenticationTools.getInforContext();
+			String gridFieldLotCode;
+			String gridFieldDescriptionCode = "2175";
+
+			if (transaction.equals("ISSUE")) {
+				gridRequest = new GridRequest("LVIRLOT", GridRequest.GRIDTYPE.LOV);
+				gridFieldLotCode = "825";
+
+				gridRequest.addParam("bin_code", bin);
+				gridRequest.addParam("part_code", part);
+				gridRequest.addParam("part_org", context.getOrganizationCode());
+				gridRequest.addParam("store_code", store);
+
+				if (requireAvailableQty) {
+					gridRequest.addFilter("availableqty", "0", ">", GridRequestFilter.JOINER.AND);
+				}
+
+			} else if (transaction.equals("RETURN")) {
+				List<Pair> udsLots = sharedPlugin.getUdsLots(part, inforClient, context);
+
+				// Check whether there are user defined lots, otherwise return all lots
+				if (udsLots != null && udsLots.size() > 0) {
+					return ok(udsLots);
+				} else {
+					gridRequest = new GridRequest("LVLOT", GridRequest.GRIDTYPE.LOV);
+					gridFieldLotCode = "2174";
+					gridRequest.setRowCount(10000);
+				}
+
+			} else {
+				throw new IllegalArgumentException("Transaction type '" + transaction + "' is not valid. Use ISSUE or RETURN");
+			}
+
+			if (lot != null && !lot.isEmpty()) {
+				gridRequest.addFilter("lotcode", lot, "=");
+			}
+
+			return ok(GridTools.convertGridResultToObject(Pair.class,
+					Pair.generateGridPairMap(gridFieldLotCode, gridFieldDescriptionCode),
+					inforClient.getGridsService().executeQuery(context, gridRequest)));
+
+		} catch (InforException e) {
+			return badRequest(e);
+		} catch(Exception e) {
+			return serverError(e);
+		}
+	}
```

Comment:
No business logic in the controllers if we can avoid it

Reply:
Could you please point out the parts that are at fault? Is it about the uds, the issue/return checking, other?

Reply:
Controllers should usually just call a method on another layer and handle the arguments/exceptions and wrap the response to send.

Reply:
I now see what you mean, but I'll need some confirmation on how to best to do it. Should I create another class alongside this one, say `PartUsage`? And there put the business logic in methods such as `loadLotListIssue` and `loadLotListReturn`?